third_party/docfx-doclet-143274/src/main/java/com/microsoft/build/BuilderUtil.java

```diff
@@ -51,21 +59,29 @@ static String populateUidValues(String text, LookupContext lookupContext) {
             }
 
             String linkContent = linkContentMatcher.group();
-            String uid = resolveUidFromLinkContent(linkContent, lookupContext);
+            String uid = resolveUidFromLinkContent(linkContent, packageName, lookupContext);
             String updatedLink = linkContentMatcher.replaceAll(uid);
             text = StringUtils.replace(text, link, updatedLink);
         }
         return text;
     }
 
     /**
+     *
      * The linkContent could be in following format
-     * #memeber
-     * Class#member
-     * Class#method()
-     * Class#method(params)
+     * 1. #member
+     * 2. Class#member
+     * 3. Class#method()
+     * 4. Class#method(params)
+     * 5. Package.Class# +{Any combination of above}
+     * All Possibilities listed: https://docs.oracle.com/javase/7/docs/technotes/tools/windows/javadoc.html#link
+     *
+     * @param linkContent Text of the link
+     * @param packageName MetaFileItem's packageName
+     * @param lookupContext LookupContext
+     * @return Fully qualified link url
      */
-    static String resolveUidFromLinkContent(String linkContent, LookupContext lookupContext) {
+    static String resolveUidFromLinkContent(String linkContent, String packageName, LookupContext lookupContext) {
         if (StringUtils.isBlank(linkContent)) {
```

Comment:
Do you think it's possible not to add the additional parameter and leveraging `lookupContext`? For me, right now, it seems that the bug comes from lookupContext. I see the variable holds the fully-qualified class name for symbols.

<img width="1339" alt="Screen Shot 2022-06-17 at 4 34 17 PM" src="https://user-images.githubusercontent.com/28604/174397983-b6f3424b-0c2e-4c51-ad74-44894c1e93e0.png">